app/controllers/agentAppointment/AppointAgentController.scala

```diff
@@ -279,6 +257,33 @@ class AppointAgentController @Inject()(
       )
   }
 
+  private def appointAgentPropertiesBadRequest(errors: Form[_])(implicit request: BasicAuthenticatedRequest[_]) = {
+    val data: Map[String, String] = errors.data
+    accounts.withAgentCode(data("agentCode")).flatMap {
```

Comment:
Is ```data("agentCode")``` guaranteed not to throw NoSuchElementException here?

Reply:
yes, agentCode is a hidden field in both forms on the page